components/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt/src/main/java/org/wso2/carbon/identity/api/resource/mgt/dao/impl/CacheBackedApiResourceMgtDAO.java

```diff
@@ -0,0 +1,284 @@
+    }
+
+    @Override
+    public List<ApiResource> getApiResources(Integer limit, Integer tenantId, String sortOrder,
+                                             List<ExpressionNode> expressionNodes)
+            throws ApiResourceManagementException {
+
+        return apiResourceManagementDAO.getApiResources(limit, tenantId, sortOrder, expressionNodes);
+    }
+
+    @Override
+    public Integer getApiResourcesCount(Integer tenantId, List<ExpressionNode> expressionNodes)
+            throws ApiResourceManagementException {
+
+        return apiResourceManagementDAO.getApiResourcesCount(tenantId, expressionNodes);
+    }
+
+    @Override
+    public List<Scope> getScopesByApi(String apiId, Integer tenatId) throws ApiResourceManagementServerException {
+
+        return apiResourceManagementDAO.getScopesByApi(apiId, tenatId);
+    }
+
+    @Override
+    public ApiResource addApiResource(ApiResource apiResource, Integer tenantId) throws ApiResourceManagementException {
+
+        return apiResourceManagementDAO.addApiResource(apiResource, tenantId);
+    }
+
+    @Override
+    public boolean isApiResourceExist(String identifier, Integer tenantId) throws ApiResourceManagementException {
+
+        return apiResourceManagementDAO.isApiResourceExist(identifier, tenantId);
+    }
+
+    @Override
+    public boolean isApiResourceExistById(String apiId, Integer tenantId) throws ApiResourceManagementException {
+
+        return apiResourceManagementDAO.isApiResourceExistById(apiId, tenantId);
+    }
+
+    @Override
+    public ApiResource getApiResourceById(String apiId, Integer tenantId) throws ApiResourceManagementException {
+
+        ApiResourceIdCacheKey cacheKey = new ApiResourceIdCacheKey(apiId);
+        ApiResourceCacheEntry entry = apiResourceCacheById.getValueFromCache(cacheKey, tenantId);
+
+        if (entry != null) {
+            log.debug("Cache entry found for API Resource " + apiId);
+            return entry.getApiResource();
+        } else {
+            log.debug("Cache entry not found for API Resource " + apiId + ". Fetching entry from DB");
+        }
+
+        ApiResource apiResource = apiResourceManagementDAO.getApiResourceById(apiId, tenantId);
+
+        if (apiResource != null) {
+            log.debug("Entry fetched from DB for API Resource " + apiId + ". Updating cache");
+            apiResourceCacheById.addToCache(cacheKey, new ApiResourceCacheEntry(apiResource), tenantId);
+            if (apiResource.getIdentifier() != null) {
+                ApiResourceIdentifierCacheKey apiResourceIdentifierCacheKey = new ApiResourceIdentifierCacheKey(
+                        apiResource.getIdentifier());
+                apiResourceCacheByIdentifier.addToCache(apiResourceIdentifierCacheKey,
+                        new ApiResourceCacheEntry(apiResource), tenantId);
+            }
+        } else {
+            log.debug("Entry for API Resource " + apiId + " not found in cache or DB");
+        }
```

Comment:
Wrapp this debug log with isDebug log enable check

Reply:
Fix this in all other places

Reply:
https://github.com/wso2/carbon-identity-framework/pull/4854/commits/8dd16447d6000412e3ff2dd1b2a9ed78249f2d13